commands/operator-sdk/cmd/scorecard/test_definitions.go

```diff
@@ -0,0 +1,277 @@
+// Copyright 2019 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package scorecard
+
+import (
+	"context"
+
+	olmapiv1alpha1 "github.com/operator-framework/operator-lifecycle-manager/pkg/api/apis/operators/v1alpha1"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// Type Definitions
+
+type Test interface {
+	GetName() string
+	GetDescription() string
+	Run(context.Context) *TestResult
+}
+
+type TestResult struct {
+	Test          Test
+	EarnedPoints  int
+	MaximumPoints int
+	Suggestions   []string
+	Errors        []error
+	isCumulative  bool
+}
+
+type TestInfo struct {
+	Name        string
+	Description string
+}
+
+// Any struct that embeds TestInfo only needs to
+// implement Run to implement the Test interface
+func (i TestInfo) GetName() string        { return i.Name }
+func (i TestInfo) GetDescription() string { return i.Description }
+
+type BasicTestConfig struct {
+	Client        client.Client
+	CR            *unstructured.Unstructured
+	ProxyPod      *v1.Pod
+	Timeout       int
+	RetryInterval int
+}
+
+type OLMTestConfig struct {
+	Client   client.Client
+	CR       *unstructured.Unstructured
+	CSV      *olmapiv1alpha1.ClusterServiceVersion
+	CRDsDir  string
+	ProxyPod *v1.Pod
```

Comment:
Same question about having the `ProxyPod` field in the `OLMTestConfig` struct.  I don't see where we're using it. Are we planning on using it in the future?

Reply:
We will need it when PR #1027 gets merged